laiaaa0/assignment3/TestingFunctions.cpp

```diff
@@ -11,94 +10,70 @@ void CreateSampleGrid(std::vector<std::vector<char> > &grid){
 }
 
 
-Dictionary * test_CreateDictionaryFromVector(){
-
+Dictionary test_CreateDictionaryFromVector(){
     std::vector<std::string> v{"car","card","cart","cat","rat"};
-    Dictionary * d = new Dictionary(v);
+    Dictionary d(v);
     return d;
-
 }
-void test_IsWordInDictionary(Dictionary * d){
-  EXPECT_TRUE(d->IsWord("cat"));
-  EXPECT_TRUE(d->IsWord("car"));
-  EXPECT_TRUE(d->IsWord("rat"));
-  EXPECT_FALSE(d->IsWord("c"));
-  EXPECT_FALSE(d->IsWord("RAT"));
-  EXPECT_FALSE(d->IsWord("ca"));
 
+void test_IsWordInDictionary(Dictionary & d){
+  EXPECT_TRUE(d.IsWord("cat"));
+  EXPECT_TRUE(d.IsWord("car"));
+  EXPECT_TRUE(d.IsWord("rat"));
+  EXPECT_FALSE(d.IsWord("c"));
+  EXPECT_FALSE(d.IsWord("RAT"));
+  EXPECT_FALSE(d.IsWord("ca"));
 }
-void test_IsPrefixInDictionary (Dictionary * d){
-  EXPECT_TRUE(d->IsPrefix("c"));
-  EXPECT_TRUE(d->IsPrefix("ca"));
-  EXPECT_TRUE(d->IsPrefix("cat"));
-  EXPECT_TRUE(d->IsPrefix("card"));
-  EXPECT_FALSE(d->IsPrefix(""));
-  EXPECT_FALSE(d->IsPrefix("a"));
-  EXPECT_FALSE(d->IsPrefix("Ra"));
 
+void test_IsPrefixInDictionary (Dictionary & d){
+  EXPECT_TRUE(d.IsPrefix("c"));
+  EXPECT_TRUE(d.IsPrefix("ca"));
+  EXPECT_TRUE(d.IsPrefix("cat"));
+  EXPECT_TRUE(d.IsPrefix("card"));
+  EXPECT_FALSE(d.IsPrefix(""));
+  EXPECT_FALSE(d.IsPrefix("a"));
+  EXPECT_FALSE(d.IsPrefix("Ra"));
 }
 
-void test_IsWordAfterRemoval (Dictionary * d){
-  EXPECT_TRUE(d->RemoveWord("cat"));
-  EXPECT_FALSE(d->IsWord("cat"));
-
+void test_IsWordAfterRemoval (Dictionary & d){
+  EXPECT_TRUE(d.RemoveWord("cat"));
+  EXPECT_FALSE(d.IsWord("cat"));
 }
-void test_IsPrefixAfterRemoval (Dictionary * d){
-  EXPECT_TRUE(d->RemoveWord("card"));
-  EXPECT_TRUE(d->IsPrefix("c"));
-  EXPECT_TRUE(d->IsPrefix("ca"));
-  EXPECT_TRUE(d->IsPrefix("car"));
-  EXPECT_FALSE(d->IsPrefix("card"));
-
-  EXPECT_TRUE(d->RemoveWord("rat"));
-  EXPECT_FALSE(d->IsPrefix("r"));
-  EXPECT_FALSE(d->IsPrefix("ra"));
-  EXPECT_FALSE(d->IsPrefix("rat"));
-
-
 
+void test_IsPrefixAfterRemoval (Dictionary & d){
```

Comment:
Ideally, tests have no parameters, they are encapsulated and there is no state going in/out.
This makes them easier to read and it should be possible to execute each test in isolation (imagine you are fixing a failing test and all tests take a long time, you will want to run only the failing test while you iterate, and all again afterwards!!).